src/Serialization/main.jl

```diff
@@ -71,59 +71,12 @@ macro registerSerializationType(ex::Any, str::Union{String,Nothing} = nothing)
-    AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.Frac{fmpq_poly}} => "AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.Frac{fmpq_poly}}",
-    AbstractAlgebra.Generic.Poly{Hecke.NfRelElem{nf_elem}} => "AbstractAlgebra.Generic.Poly{Hecke.NfRelElem{nf_elem}}",
-    AbstractAlgebra.Generic.Poly{Hecke.NfRelNSElem{nf_elem}} => "AbstractAlgebra.Generic.Poly{Hecke.NfRelNSElem{nf_elem}}",
-    AbstractAlgebra.Generic.Poly{nf_elem} => "AbstractAlgebra.Generic.Poly{nf_elem}",
-    AbstractAlgebra.Generic.Poly{NfAbsNSElem} => "AbstractAlgebra.Generic.Poly{NfAbsNSElem}",
-    AbstractAlgebra.Generic.Poly{padic} => "AbstractAlgebra.Generic.Poly{padic}",
-    AbstractAlgebra.Generic.PolyRing{AbstractAlgebra.Generic.Frac{fmpq_poly}} => "AbstractAlgebra.Generic.PolyRing{AbstractAlgebra.Generic.Frac{fmpq_poly}}",
-    AbstractAlgebra.Generic.PolyRing{Hecke.NfRelElem{nf_elem}} => "AbstractAlgebra.Generic.PolyRing{Hecke.NfRelElem{nf_elem}}",
-    AbstractAlgebra.Generic.PolyRing{Hecke.NfRelNSElem{nf_elem}} => "AbstractAlgebra.Generic.PolyRing{Hecke.NfRelNSElem{nf_elem}}",
-    AbstractAlgebra.Generic.PolyRing{nf_elem} => "AbstractAlgebra.Generic.PolyRing{nf_elem}",
-    AbstractAlgebra.Generic.PolyRing{NfAbsNSElem} => "AbstractAlgebra.Generic.PolyRing{NfAbsNSElem}",
-    AbstractAlgebra.Generic.PolyRing{padic} => "AbstractAlgebra.Generic.PolyRing{padic}",
-    Hecke.NfRel{nf_elem} => "Hecke.NfRel{nf_elem}",
-    Hecke.NfRelElem{nf_elem} => "Hecke.NfRelElem{nf_elem}",
-    Hecke.NfRelNSElem{nf_elem} => "Hecke.NfRelNSElem{nf_elem}",
-    Matrix{fmpq_poly} => "Matrix{fmpq_poly}",
-    Hecke.NfRelNS{Hecke.NfRelElem{nf_elem}} => "Hecke.NfRelNS{Hecke.NfRelElem{nf_elem}}",
-    Hecke.NfRelNSElem{Hecke.NfRelElem{nf_elem}} => "Hecke.NfRelNSElem{Hecke.NfRelElem{nf_elem}}",
-    MPolyIdeal{AbstractAlgebra.Generic.MPoly{padic}} => "MPolyIdeal{AbstractAlgebra.Generic.MPoly{padic}}",
-    MPolyIdeal{AbstractAlgebra.Generic.MPoly{AbstractAlgebra.Generic.Frac{fmpq_poly}}} => "MPolyIdeal{AbstractAlgebra.Generic.MPoly{AbstractAlgebra.Generic.Frac{fmpq_poly}}}",
-    MPolyIdeal{AbstractAlgebra.Generic.MPoly{Hecke.NfRelElem{nf_elem}}} => "MPolyIdeal{AbstractAlgebra.Generic.MPoly{Hecke.NfRelElem{nf_elem}}}",
-    MPolyIdeal{AbstractAlgebra.Generic.MPoly{Hecke.NfRelNSElem{nf_elem}}} => "MPolyIdeal{AbstractAlgebra.Generic.MPoly{Hecke.NfRelNSElem{nf_elem}}}",
-    MPolyIdeal{AbstractAlgebra.Generic.MPoly{nf_elem}} => "MPolyIdeal{AbstractAlgebra.Generic.MPoly{nf_elem}}",
-    MPolyIdeal{AbstractAlgebra.Generic.MPoly{NfAbsNSElem}} => "MPolyIdeal{AbstractAlgebra.Generic.MPoly{NfAbsNSElem}}",
-    MPolyIdeal{fmpq_mpoly} => "MPolyIdeal{fmpq_mpoly}",
-    MPolyIdeal{fmpz_mpoly} => "MPolyIdeal{fmpz_mpoly}",
-    MPolyIdeal{fq_nmod_mpoly} => "MPolyIdeal{fq_nmod_mpoly}",
-    MPolyIdeal{nmod_mpoly} => "MPolyIdeal{nmod_mpoly}",
-    NfRelNS{nf_elem} => "NfRelNS{nf_elem}",
     Polymake.BigObjectAllocated => "Polymake.BigObject",
```

Comment:
Nice!! So perhaps this one last line should also be removed, replace with a `registerSerializationType ` call in `src/Serialization/polymake.jl` ?